tests/monster_test_generated.js

```diff
@@ -28,19 +28,19 @@ MyGame.OtherNameSpace = MyGame.OtherNameSpace || {};
  * @enum
  */
 MyGame.Example.Color = {
-  Red: 1,
-  Green: 2,
-  Blue: 8
+  Red: 1, 1: 'Red',
```

Comment:
Should these maybe be in a seperate table? I can imagine JS gets optimized better if theres only one type of key in a map. Then again performance may not be a concern here?

Reply:
That's an interesting question.

I did a quick benchmark here of objects with mixed key types which showed that integer lookups were slow, but it didn't seem to make a *huge* difference if the object had mixed types or not. http://jsben.ch/7KeeD

It's also possible this isn't a representative test case. :)

The [Typescript Enum documentation](https://www.typescriptlang.org/docs/handbook/enums.html) talks about one way their compiler "could" generate the reverse mappings, but doesn't offer guarantees:
> TypeScript might compile this down to something like the the following JavaScript:
```js
var Enum;
(function (Enum) {
    Enum[Enum["A"] = 0] = "A";
})(Enum || (Enum = {}));
var a = Enum.A;
var nameOfA = Enum[a]; // "A"
```

> In this generated code, an enum is compiled into an object that stores both forward (name -> value) and reverse (value -> name) mappings. References to other enum members are always emitted as property accesses and never inlined.

Reply:
Well, is it safer then to generate two tables? Anyone else an opinion?

Reply:
I don't think I know enough about Javascript to have an opinion. The fact that Typescript includes both mappings in a single object makes me think that it is hopefully not a huge issue.

The benchmark I did (for what it's worth) makes it look like accessing values by integer index was slow, regardless of if the object had keys of mixed types or not.

Are there some people more familiar with Javascript who we could tag here to have them weigh in?

Reply:
I asked a JS/V8 expert, who said mixing types like this is unlikely a performance issue. So we can merge as-is.